core/src/test/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTaskSpec.scala

```diff
@@ -37,6 +37,18 @@ class BackgroundTaskSpec extends AnyFlatSpec with WelderTestSuite {
     res.toString shouldBe s"gs://testBucket/notebooks/test.Rmd"
   }
 
+  "shouldSync" should "return true if passed a supported pattern" in {
+    val storageLink =
+      StorageLink(
+        LocalBaseDirectory(RelativePath(Paths.get(""))),
+        None,
+        CloudStorageDirectory(GcsBucketName("testBucket"), Some(BlobPath("notebooks"))),
+        ".*\\.R".r
+      )
+    val res = initBackgroundTask(Map(storageLink.localBaseDirectory.path -> storageLink), Map.empty, None).shouldSync(storageLink.pattern.toString())
+    res shouldBe true
```

Comment:
this looks pretty good..just realize BackgroundTask is a class....to make testing a bit easier, I would put `shouldSync` in `BackgroundTask` object.

So in `[BackgroundTask.scala](https://github.com/DataBiosphere/welder/pull/80/files#diff-4a38dd3f6392b9fe795b64ff8746380bb5c0ce340986c04ce2dc0bcb3ac8207e)` file,

```
object BackgroundTask {
  val filesShouldSyncRegex: Regex = ".*(.R)$|(.Rmd)$".r
  def shouldSync(storageLinkPattern: String): Boolean = filesShouldSyncRegex.findFirstIn(storageLinkPattern).isDefined
}  
```


This way, in test, you can just do `BackgroundTask.shouldSync` to access it.

in I would add a few more tests
like
```
BackgroundTask.shouldSync("da.R") shouldBe true
BackgroundTask.shouldSync("da.Rdd") shouldBe false
BackgroundTask.shouldSync("da.Rmd") shouldBe true
...
```